### PR TITLE
21417 refactor to put group admin endpoints under one folder

### PR DESCRIPTION
**Pull request by eason-pan-bc**
*Issue #:* /bcgov/entity#21417

*Description of changes:*
- Create admin folder
- Move following files under new admin folder (administrative_bn.py, configuration.py, dissolution.py)
- Update code such that all admin endpoints share a base admin blueprint.
- Update tests as appropriate
- Verify all admin endpoints still work after updates
 
*Local tests & endpoints verification screenshots*
- Passed all unit tests under _tests/unit/resources/v2_ after updates
image
- Create bn request endpoint
Pasted image 20240527113751
- Get configurations endpoint
Pasted image 20240527114945
- Update configurations endpoint
Pasted image 20240527115751
- Get dissolutions statistics endpoint
Pasted image 20240527120017

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/src/legal_api/resources/constants.py
Comment (eason-pan-bc) [reply legal-api/src/legal_api/resources/constants.py]: good point, fixed it.
It's now matching the existing pattern.

**Review by argush3 — APPROVED** [review]
looks good.  let's wait until after sprint demos before we merge this.

File: legal-api/src/legal_api/resources/v2/admin/configuration.py
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/admin/configuration.py:42]: @argush3 any staff can update configuration?

Comment (eason-pan-bc) [reply legal-api/src/legal_api/resources/v2/admin/configuration.py:42]: sure, gonna confirm with Dylan, will keep you posted

Comment (vysakh-menon-aot) [reply legal-api/src/legal_api/resources/v2/admin/configuration.py:42]: we have a role `UserRoles.admin_edit`, which we use in admin page. This will be a staff with an additional admin_edit role (https://github.com/bcgov/lear/blob/main/legal-api/src/legal_api/resources/v2/administrative_bn.py#L34)

Comment (eason-pan-bc) [reply legal-api/src/legal_api/resources/v2/admin/configuration.py:42]: Dylan just confirmed that any staff can update dissolution configuration

**Pull request merged**
